src/platform-implementation-js/namespaces/app-menu.ts

```diff
@@ -102,11 +104,85 @@ Items with CollapsiblePanels can also have accessories which provide primary act
 */
 export default class AppMenu {
   #driver;
+  #events = stream<
+    {
+      type: 'collapseToggled';
+      /** Whether or not the AppMenuBurger is open or not. */
+      open: boolean;
+      /**
+       * 'start' is when a `transactionstart` event is fired on a collapsible panel.
+       * 'end' is when a `transactioncancel` event is fired on a collapsible panel. */
+      stage: 'start' | 'end';
+    },
+    unknown
+  >((emitter) => {
+    const f = async () => {
+      const isShown = await this.isShown();
+
+      if (!isShown) {
+        return;
+      }
+
+      const appMenu = await GmailElementGetter.getAppMenuAsync();
+
+      if (!appMenu) {
+        return;
+      }
+
+      const onTransition =
+        (stage: 'start' | 'end') =>
+        async ({ target }: TransitionEvent) => {
+          const { TOGGLE_OPEN_STATE } = CollapsiblePanelView.elementCss;
+
+          if (
+            target instanceof HTMLElement &&
+            target.matches(CollapsiblePanelView.elementSelectors.NATIVE) &&
+            ((stage === 'start' &&
+              target.classList.contains(TOGGLE_OPEN_STATE)) ||
+              // TOGGLE_OPEN_STATE is removed on transitioncancel TransitionEvents
+              stage === 'end')
+          ) {
+            emitter.emit({
+              type: 'collapseToggled',
+              open: this.isMenuCollapsed(),
+              stage,
```

Comment:
The `open` value might get stale when this event gets triggered and handled. Personally, I'd emit just an event type and leave the value retrieval to the end user (via app menu API)

Reply:
> emit just an event type

Do you mean a `TransitionEvent`?

Reply:
[079ff95](https://github.com/InboxSDK/InboxSDK/pull/936/commits/079ff957cebe68b578391319dd4e2ab8a3d74dc6) makes checking `AppMenu.prototype.isMenuOpen` the subscriber's responsibility.